atomkraft/chain/utils.py

```diff
@@ -60,9 +60,9 @@ def __init__(
         """
         self.params: Dict[str, str] = params
         if filter:
-            self.filter: Callable[[Any], bool] = filter
+            self.filter: Optional[Callable[[Any], bool]] = filter
         else:
-            self.filter = lambda x: True
+            self.filter = None
```

Comment:
Thanks @ivan-gavran for pointing this out.

```suggestion
        self.filter: Optional[Callable[[Any], bool]] = filter
```